src/Test/TestGameEvent.java

```diff
@@ -244,11 +244,7 @@ boolean testMoveEvent()
     testPassed &= validate(7 == mech.x && 6 == mech.y, "    Mech should think he is at (7, 6) after moving.");
     testPassed &= validate(testMap.getLocation(7, 6).getResident() == mech, "    Mech is not at (7, 6) after moving.");
 
-    path.addWaypoint(7, 0); // New endpoint over water.
-    new MoveEvent(mech, path).performEvent(testMap); // This should not execute. Water is bad for grunts.
-    testPassed &= validate(7 == mech.x && 6 == mech.y, "    Mech does not think he is at (7, 6), but should.");
-    testPassed &= validate(testMap.getLocation(7, 6).getResident() == mech, "    Mech is not still at (7, 6), but should be.");
-    testPassed &= validate(testMap.getLocation(7, 0).getResident() == null, "    Location (7, 0) should still be empty.");
+    // Test moving into water removed since MoveEvent no longer validates that the unit can actually get there.
```

Comment:
Huh. I misunderstood when we were talking before. I thought you were saying we had a test case that was *accidentally* trying to move a unit over water. I thought the fix for the "broken test case" would be to change the destination so it was a valid move.
Since this test is doing it on purpose to validate that you can't do that, we should probably keep it around (and re-add the corresponding validity checks to `MoveEvent`).
I can do that if you want since I'm the one who inadvertently told you to tear it out.

Reply:
I don't really care that much; as noted, the check not being there doesn't really break anything (as MoveEvents are generated by a process which *does* validate)

Reply:
I figure an equivalent test could also be constructed using `enqueueMoveEvent()`

Reply:
Sure, but it's also possible to construct a MoveEvent without going through `enqueueMoveEvent()`.
I mean, we probably can live without this test and the checks that make it pass, but I hate to remove a valid test case.